File: web/src/app/core/model/thread/thread-response.ts
import {PublicMetricsResponse} from './public-metrics-response';
import {HashtagResponse} from './hashtag-response';
import {UrlEntityResponse} from './url-entity-response';
import {UserResponse} from './user-response';
import {MediaResponse} from './media-response';
import {InteractionContextResponse} from './interaction-context-response';

export interface ThreadResponse {
  id: string;
  content: string;
  createdAt: string;
  replySettings: string;
  lang: string;
  publicMetrics: PublicMetricsResponse;
  hashtags: HashtagResponse[];
  urls: UrlEntityResponse[];
  media: MediaResponse[];
  author: UserResponse;
  interactions: InteractionContextResponse
}

File: core/thread/src/main/java/com/example/thread/dto/response/ThreadResponse.java
package com.example.thread.dto.response;

import java.time.Instant;
import java.util.List;

public record ThreadResponse(
        String id,
        String text,
        Instant createdAt,
        String replySettings,
        String lang,
        PublicMetricsResponse publicMetrics,
        List<HashtagResponse> hashtags,
        List<UrlEntityResponse> urls,
        List<MediaResponse> media,
        UserResponse author
) {}